src/handlers/strategies/directoryListing.ts

```diff
@@ -176,8 +177,18 @@ async function fetchR2Result(
     }
   }
 
-  // R2 isn't having a good day, return a 500
-  throw new Error(`R2 failed listing path ${bucketPath}: ${r2Error}`);
+  // R2 isn't having a good day, log to sentry & rewrite to direct.nodejs.org
+  const error = new Error(`R2 failed listing path ${bucketPath}: ${r2Error}`);
+  if (ctx.env.USE_FALLBACK_WHEN_R2_FAILS) {
```

Comment:
I wonder if we can avoid calling fallback in case fallback also failed previously.
will these requests be cached?

Reply:
As of now they're not cached. In my opinion at least if all of the R2 requests fail, and then the request to the fallback server fails, we can just return a 500 or 503.

If we wanted to for some extra resiliency, instead of falling back to `direct.nodejs.org` we could move the current Cloudflare load balancer config to another subdomain (e.g. `r2-fallback.nodejs.org`) which tries to go to the DO server first, but if it's unhealthy it will point to the backup server.

Reply:
> If we wanted to for some extra resiliency, instead of falling back to direct.nodejs.org we could move the current Cloudflare load balancer config to another subdomain (e.g. r2-fallback.nodejs.org) which tries to go to the DO server first, but if it's unhealthy it will point to the backup server.

Indeed, since we cannot use Load Balancers with Workers for the main domain, we could create a subdomain "fallback.nodejs.org" which is behind a Load Balancer (The DO server and Backup Server) I think that'd be better than calling "direct.nodejs.org"

Can we do that? cc @targos as we would need to setup a subdomain and a load balancer there.

Reply:
Don't we already have a load balancer on https://origin.nodejs.org ?

Reply:
Oh yes, we do. @flakey5 do you think we can make it to go to origin.nodejs.org?

Reply:
Added!